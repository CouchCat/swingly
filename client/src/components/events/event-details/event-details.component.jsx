import React from 'react';
import {
  EventDetailsContainer,
  EventImageContainer,
  EventImage,
  EventTitle,
  DetailsContainer,
  ButtonsContainer,
  VerticalDivider,
  EventButton,
} from './event-details.styles';
import Modal from '../../../ui/modal/modal.component';
import CloseButton from '../../../ui/button/button-close/button-close.component';
import Details from './details/details.component';
import Backdrop from '../../../ui/backdrop/backdrop.component';

export const EventDetails = (props) => {
  const { isOpen, onClose, event } = props;

  return (
    <>
      <Backdrop isOpen={isOpen} onClick={onClose}/>
      <Modal
        isOpen={isOpen}
        transName="event-details">
        { event &&
          <EventDetailsContainer>
            <CloseButton onClick={onClose} />
            <EventImageContainer>
              <EventImage src='http://lorempixel.com/400/200/cats' />
            </EventImageContainer>
            <EventTitle>
              {event.title}
            </EventTitle>
            <DetailsContainer>
              <Details event={event} />
            </DetailsContainer>
            <ButtonsContainer>
              <EventButton transparent>Guests</EventButton>
              <VerticalDivider />
              <EventButton transparent>Interested</EventButton>
              <VerticalDivider />
              <EventButton transparent>Going</EventButton>
            </ButtonsContainer>
          </EventDetailsContainer>
        }
      </Modal>
    </>
  );
}


export default EventDetails;
