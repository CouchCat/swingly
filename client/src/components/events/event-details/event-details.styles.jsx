import styled from 'styled-components';
import Button from '../../../ui/button/button.component';

export const EventDetailsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 80vw;
`;
EventDetailsContainer.displayName = 'EventDetailsContainer';

export const EventImageContainer = styled.div`
  display: flex;
  height: 10rem;
  width: 120%;
  margin: -2rem -2rem 0 -2rem;
  background-color: #fff;
  border-radius: 5px;
`;
EventImageContainer.displayName = 'EventImageContainer';

export const EventImage = styled.img`
  object-fit: cover;
  width: 100%;
  border-radius: 5px 5px 0 0;
`;
EventImage.displayName = 'EventImage';

export const EventTitle = styled.div`
  margin: 2rem 0 1rem 0;
  width: 100%;
  text-align: center;
  ${props => props.theme.typo.h5};
  color: ${props => props.theme.palette.text.primary};
`;
EventTitle.displayName = 'EventTitle';

export const DetailsContainer = styled.div`
  margin: 1rem 0 1rem 0;
  width: 100%;
`;
DetailsContainer.displayName = 'DetailsContainer';

export const ButtonsContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
`;
ButtonsContainer.displayName = 'ButtonsContainer';

export const EventButton  = styled(Button)`
  flex: 1 1 100%;
  ${props => props.theme.typo.body};
  margin: 0;
  color: ${props => props.theme.palette.text.primary};
  box-shadow: none;

  :hover {
    box-shadow: none;
    color: ${props => props.theme.palette.primary.main};
  }
`;
EventButton.displayName = 'EventButton';

export const VerticalDivider = styled.div`
  margin: auto 1.5rem auto 1.5rem;
  height: 70%;
  border-right: 1px solid ${props => props.theme.palette.grey[5]};
`;
VerticalDivider.displayName = 'VerticalDivider';

