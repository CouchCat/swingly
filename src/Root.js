import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';

import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
// import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './redux/root-reducer';
import { library } from '@fortawesome/fontawesome-svg-core';
import { 
  faBars,
  faSyncAlt,
  faUserCircle,
  faEdit,
  faSignOutAlt,
  faSignInAlt,
  faHome,
  faCalendar,
  faCalendarDay,
  faCalendarWeek,
  faEllipsisV,
  faCalendarPlus,
  faHatWizard,
  faMoon,
  faSun,
  faChevronLeft
} from '@fortawesome/free-solid-svg-icons';

export default (props) => {
  const { children, initialState = {} } = props;
  const middlewares = [thunk];

  if (process.env.NODE_ENV === 'development') {
    middlewares.push(logger);
  }
  
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middlewares),
  );

  const persistor = persistStore(store);

  library.add(
    faBars,
    faSyncAlt,
    faUserCircle,
    faEdit,
    faSignOutAlt,
    faSignInAlt,
    faHome,
    faCalendar,
    faCalendarDay,
    faCalendarWeek,
    faEllipsisV,
    faCalendarPlus,
    faHatWizard,
    faMoon,
    faSun,
    faChevronLeft
  );

  return (
    <Provider store={store}>
      <BrowserRouter>
        <PersistGate persistor={persistor}>
          {children}
        </PersistGate>
      </BrowserRouter>
    </Provider>
  );
}