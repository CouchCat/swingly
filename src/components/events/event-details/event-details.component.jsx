import React from 'react';
import {
  EventDetailsContainer,
  EventImageContainer,
  EventImage,
  EventTitle,
  Details,
  Row,
  Label,
  EventData
} from './event-details.styles'
import Modal from '../../../ui/modal/modal.component';
import CloseButton from '../../../ui/button/button-close/button-close.component';

export const EventDetails = (props) => {
  const { isOpen, onClose } = props;

  return (
    <Modal
      isOpen={isOpen}
      direction="left"
      transName="events-details">
      <EventDetailsContainer>
        <CloseButton onClick={onClose} />
        <EventImageContainer>
          <EventImage src='http://lorempixel.com/400/200/cats'  />
        </EventImageContainer>
        <EventTitle>
          Swingtanzen macht froh!
        </EventTitle>
        <Details>
          <Row>
            <Label>Location</Label>
            <EventData>Sunny-side-of-the-street 42.</EventData>
          </Row>
          <Row>
            <Label>Location</Label>
            <EventData>Sunny-side-of-the-street 42.</EventData>
          </Row>
          <Row>
            <Label>Location</Label>
            <EventData>Sunny-side-of-the-street 42.</EventData>
          </Row>

        </Details>
      </EventDetailsContainer>
    </Modal>
  );
}

export default EventDetails;