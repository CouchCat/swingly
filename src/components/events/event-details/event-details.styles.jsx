import styled from 'styled-components';

export const EventDetailsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  min-width: 26rem;
`;
EventDetailsContainer.displayName = 'EventDetailsContainer';

export const EventImageContainer = styled.div`
  display: flex;
  height: 10rem;
  width: 120%;
  margin: -2rem -2rem 0 -2rem;
  background-color: #fff;
`;
EventImageContainer.displayName = 'EventImageContainer';

export const EventImage = styled.img`
  object-fit: cover;
  width: 100%;
`;
EventImage.displayName = 'EventImage';

export const EventTitle = styled.div`
  margin: 2rem 0 1rem 0;
  width: 100%;
  text-align: center;
  ${props => props.theme.typo.h5};
  color: ${props => props.theme.palette.text.primary};
`;
EventTitle.displayName = 'EventTitle';

export const Details = styled.div`
  margin: 1rem 0 1rem 0;
  width: 100%;
`;
Details.displayName = 'Details';

export const Row = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;
Row.displayName = 'Row';

export const Label = styled.div`
  flex: 1 1 20%;

`;
Label.displayName = 'Label';

export const EventData = styled.div`
  flex: 1 1 80%;

`;
EventData.displayName = 'EventData';