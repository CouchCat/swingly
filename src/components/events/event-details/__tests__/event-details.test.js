import React from 'react';
import { EventDetails } from '../event-details.component';
import { shallow, mount } from 'enzyme';
import Root from '../../../../Root';
import renderer from 'react-test-renderer';

describe('Event Details Component', () => {
  // let wrapper;

  it('should render correctly', () => {
    const tree = renderer.create(<EventDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});