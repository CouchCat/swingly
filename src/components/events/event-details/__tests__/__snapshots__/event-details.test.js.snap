// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Details Component should render correctly 1`] = `null`;
